src/core/Details/Details.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { forwardRef, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { baseStyles } from './Details.baseStyles';
+import { SuomifiThemeConsumer, SuomifiThemeProp } from '../theme';
+import {
+  spacingStyles,
+  separateMarginProps,
+  MarginProps,
+} from '../theme/utils/spacing';
+import {
+  HtmlDetails,
+  HtmlDetailsProps,
+} from '../../reset/HtmlDetails/HtmlDetails';
+import { HtmlDiv } from '../../reset';
+
+interface InternalDetailsProps extends HtmlDetailsProps, MarginProps {
+  /** CSS class for custom styles */
+  className?: string;
+  children: ReactNode;
+  /** Callback fired on summary element click */
+  onClick?: (event: React.MouseEvent) => void;
+  /** Ref object is passed to the details element. Alternative to React `ref` attribute. */
+  forwardedRef?: React.RefObject<HTMLDetailsElement>;
+  /** Label for the component. Will be used as content in the summary element */
+  summaryLabel: ReactNode;
+  /** Open status of the component for controlled state */
+  open?: boolean;
+}
+
+export type DetailsProps = InternalDetailsProps;
```

Comment:
Why not export the interface directly? Is there a need for this type?

Reply:
I was probably thinking of a past issue with using interfaces with function components. But it is redundant here 👍